vega_sim/api/data.py

```diff
@@ -1245,6 +1334,73 @@ def _liquidity_provider_fee_share_from_proto(
     ]
 
 
+def _liquidity_sla_from_proto(
+    liquidity_provider_sla: List[vega_protos.vega.LiquidityProviderSLA],
+    decimal_spec: DecimalSpec,
+) -> List[LiquiditySLA]:
+    decimals = decimal_spec.price_decimals + decimal_spec.position_decimals
+    return [
+        LiquiditySLA(
+            party=individual_liquidity_provider_sla.party,
+            current_epoch_fraction_of_time_on_book=(
+                0
+                if individual_liquidity_provider_sla.current_epoch_fraction_of_time_on_book
+                == ""
+                else float(
+                    individual_liquidity_provider_sla.current_epoch_fraction_of_time_on_book
+                )
+            ),
+            last_epoch_fraction_of_time_on_book=(
+                0
+                if individual_liquidity_provider_sla.last_epoch_fraction_of_time_on_book
+                == ""
+                else float(
+                    individual_liquidity_provider_sla.last_epoch_fraction_of_time_on_book
+                )
+            ),
+            last_epoch_fee_penalty=(
+                0
+                if individual_liquidity_provider_sla.last_epoch_fee_penalty == ""
+                else float(individual_liquidity_provider_sla.last_epoch_fee_penalty)
+            ),
+            last_epoch_bond_penalty=(
+                0
+                if individual_liquidity_provider_sla.last_epoch_bond_penalty == ""
+                else float(individual_liquidity_provider_sla.last_epoch_bond_penalty)
+            ),
+            hysteresis_period_fee_penalties=[
+                float(x)
+                for x in individual_liquidity_provider_sla.hysteresis_period_fee_penalties
+            ],
+            required_liquidity=(
+                0
+                if individual_liquidity_provider_sla.required_liquidity == ""
+                else num_from_padded_int(
+                    float(individual_liquidity_provider_sla.required_liquidity),
+                    decimals,
+                )
+            ),
```

Comment:
I'm not 100% but am trying to check now. Are the right decimals being used here?

I would have assumed it was just the asset decimals.

Reply:
Yeah, after a bit of digging it does seem to be asset decimals, changed now.